modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -12511,4 +12511,139 @@ public static long uncompressedSize(File zip) throws IOException {
     public static String enabledString(boolean enabled) {
         return enabled ? "enabled" : "disabled";
     }
+
+    /**
+     * Calculates intersection between two sets.
+     * @see #sizeOfSetIntersection
+     *
+     * @param lhs Left set.
+     * @param rhs Right sets.
+     * @param <T> Element type.
+     * @return Set which is intersection between two sets.
+     */
+    @SafeVarargs
+    public static <T> Set<T> setIntersection(Set<? extends T> lhs, Set<? extends T>... rhs) {
+        if (lhs == null || rhs == null)
+            return Collections.emptySet();
+
+        Set<T> result = new HashSet<>();
+
+        for (T val : lhs) {
+            int contains = 0;
+
+            for (Set<? extends T> rSet : rhs) {
+                if (rSet.contains(val))
+                    contains++;
+                else
+                    break;
+            }
+
+            if (contains == rhs.length)
+                result.add(val);
+        }
+
+        return result;
+    }
+
+    /**
+     * Calculates cardinality (size) of intersection between two sets.
+     * @see #setIntersection
+     *
+     * @param lhs Left set.
+     * @param rhs Right sets.
+     * @param <T> Element type.
+     * @return Size of intersection between two sets.
+     */
+    @SafeVarargs
+    public static <T> int sizeOfSetIntersection(Set<? extends T> lhs, Set<? extends T>... rhs) {
```

Comment:
I did not find any use in the core code, only in self tests - then the method is not needed.